test-cases/tests/alerts/c18-validate-api-usage-alerts.md

```diff
@@ -0,0 +1,126 @@
+
+- integreatlyOperatorBranchName: (Go to integreatly-operator github repo and select latest RHOAM release branch)
+- ocmClusterLifespan: 36
+- openshiftVersion: (search for the version in the test plan document)
+- useByoc: false
+- multiAZ: false
+- patchCloudResAwsStrCM: false
+- clusterID: api-usage-alerts
+- emailRecipients: <your@email.address>
+- stepsToDo: provision + install
+
+## Steps
+
+1. Validate that rate-limiting alerts ConfigMap has been created
+
+```
+oc get cm rate-limit-alerts -n redhat-managed-api-operator
+```
+
+2. Verify that level1, level2 and level3 rate-limiting alerts are present
+
+```
+oc get prometheusrules -n redhat-managed-api-marin3r
+```
+
+3. Modify the `sku-limits-managed-api-service` configmap to set the rate limit to 20 requests per minute.
+   Run
+
+```
+oc edit configmap sku-limits-managed-api-service -n redhat-rhmi-operator
+```
+
+And insert the following data:
+
+```
+data:
+  rate_limit: |-
+    {
+      "RHOAM SERVICE SKU": {
+        "unit": "minute",
+        "requests_per_unit": 20
+      }
+    }
+```
+
+4. Modify the `rate-limit-alerts` to allow alerts to fire in a reasonable time from the testing perspective:
+
+```
+oc edit configmap rate-limit-alerts -n redhat-rhmi-operator
+```
+
+and insert the following data:
+
+```
+{
+      "api-usage-alert-level1": {
+        "ruleName": "Level1ThreeScaleApiUsageThresholdExceeded",
+        "level": "warning",
+        "minRate": "80%",
+        "maxRate": "90%",
+        "period": "1m"
+      },
+      "api-usage-alert-level2": {
+        "ruleName": "Level2ThreeScaleApiUsageThresholdExceeded",
+        "level": "warning",
+        "minRate": "90%",
+        "maxRate": "95%",
+        "period": "2h"
+      },
+      "api-usage-alert-level3": {
+        "ruleName": "Level3ThreeScaleApiUsageThresholdExceeded",
+        "level": "warning",
+        "minRate": "95%",
+        "period": "30m"
+      }
+}
+```
+
+4. Go to `Networking` > `Routes` under `redhat-rhmi-3scale` namespace
+5. Click on the `zync` route that starts with `https://3scale-admin...`
+6. Go to `Secrets` > `system-seed` under 3Scale namespace and copy the admin password
+7. Go back to 3Scale login page and login
+8. Click on `Ok, how does 3scale work?` and follow the 3Scale wizard to create an API
+9. Once on your API overview page, click on `Integration` on the left, then on `Configuration`
+10. Copy the `example curl for testing` for `Staging-APIcast` and paste into a terminal window
+11. Run the following script to verify that rate limit (20 requests/minute) works correctly:
+
+```
+for i in {1..21}; do <replace-with-example-curl-for-testing>; done
+```
+
+> Only the last request should fail (TODO: any specific status code?)
+
+12. TODO: should we update the configmap now to increase the rate limit for test the rate-limit alerts?
```

Comment:
I think that we can test the 3 levels without modifying the rate limit configuration. I'm still trying to find the sweet spot to replicate the alerts as consistently as possible, so maybe 20 reqs/minute is not the best, but it's possible to replicate all of the levels with that value

Reply:
We could always look to set the dev limits to 100 requests per minute which would make our calculations much easier